packages/vaadin-virtual-list/src/virtualizer-iron-list-adapter.js

```diff
@@ -0,0 +1,430 @@
+    let targetVirtualIndex = Math.floor((index / this.size) * this._virtualCount);
+    if (this._virtualCount - targetVirtualIndex < visibleElementCount) {
+      targetVirtualIndex = this._virtualCount - (this.size - index);
+      this._vidxOffset = this.size - this._virtualCount;
+    } else if (targetVirtualIndex < visibleElementCount) {
+      if (index < OFFSET_ADJUST_MIN_THRESHOLD) {
+        targetVirtualIndex = index;
+        this._vidxOffset = 0;
+      } else {
+        targetVirtualIndex = OFFSET_ADJUST_MIN_THRESHOLD;
+        this._vidxOffset = index - targetVirtualIndex;
+      }
+    } else {
+      this._vidxOffset = index - targetVirtualIndex;
+    }
+
+    do {
+      this.__skipNextVirtualIndexAdjust = true;
+      super.scrollToIndex(targetVirtualIndex);
+      this._scrollHandler();
+    } while (this.firstVisibleIndex !== index - this._vidxOffset && this._scrollTop < this._maxScrollTop && !this.grid);
+  }
+
+  flush() {
+    this._resizeHandler();
+    flush();
+    this._scrollHandler();
+    this.__scrollReorderDebouncer && this.__scrollReorderDebouncer.flush();
+    this.__debouncerWheelAnimationFrame && this.__debouncerWheelAnimationFrame.flush();
+  }
+
+  update(startIndex = 0, endIndex = this.size - 1) {
+    this.__getVisibleElements().forEach((el) => {
+      if (el.__virtualIndex >= startIndex && el.__virtualIndex <= endIndex) {
+        this.updateElement(el, el.__virtualIndex);
+      }
+    });
+  }
+
+  __getIndexScrollOffset(index) {
+    const element = this.__getVisibleElements().find((el) => el.__virtualIndex === index);
+    return element ? this.scrollTarget.getBoundingClientRect().top - element.getBoundingClientRect().top : undefined;
+  }
+
+  set size(size) {
+    if (size === this.size) {
+      return;
+    }
+
+    let fvi = this.firstVisibleIndex + this._vidxOffset;
+
+    const fviOffsetBefore = this.__getIndexScrollOffset(fvi);
+
+    this.__size = size;
+    this._itemsChanged({
+      path: 'items'
+    });
+
+    fvi = Math.min(fvi, size - 1);
+    this.scrollToIndex(fvi);
+
+    const fviOffsetAfter = this.__getIndexScrollOffset(fvi);
+    if (fviOffsetBefore !== undefined && fviOffsetAfter !== undefined) {
+      this._scrollTop += fviOffsetBefore - fviOffsetAfter;
+    }
+
+    if (!this.elementsContainer.children.length) {
+      requestAnimationFrame(() => this._resizeHandler());
+    }
+  }
+
+  get size() {
+    return this.__size;
+  }
+
+  /** @private */
+  get _scrollTop() {
+    return this.scrollTarget.scrollTop;
+  }
+
+  /** @private */
+  set _scrollTop(top) {
+    this.scrollTarget.scrollTop = top;
+  }
+
+  /** @private */
+  get items() {
+    return {
+      length: Math.min(this.size, MAX_VIRTUAL_COUNT)
+    };
+  }
+
+  /** @private */
+  get offsetHeight() {
+    return this.scrollTarget.offsetHeight;
+  }
+
+  /** @private */
+  get $() {
+    return {
+      items: this.scrollContainer
+    };
+  }
```

Comment:
These properties are directly accessed from the iron-list source code. Didn't want to do modifications to the iron-list source so added the properties here in the adapter instead.